src/sm_bluesky/common/plan_stubs/set_and_do_other_plan.py

```diff
@@ -0,0 +1,47 @@
+from collections.abc import Callable
+
+import bluesky.plan_stubs as bps
+from bluesky import preprocessors as bpp
+from bluesky.plan_stubs import abs_set, sleep
+from bluesky.protocols import Movable, Readable
+from bluesky.utils import MsgGenerator, plan
+
+MR = type("MR", (Movable, Readable), {"MR": "MR"})
+
+
+def set_setpoint_to_readback(set_signal, readback_signal) -> MsgGenerator:
+    a = yield from bps.rd(readback_signal)
+    yield from bps.abs_set(set_signal, a, wait=True)
+
+
+@plan
+def set_and_wait_within_tolerance(
+    set_signal: MR,
+    value: float,
+    tolerance: float,
+    readback_signal: Readable | None = None,
+    plan: Callable | None = None,
+    plan_parm: list | None = None,
```

Comment:
I think this is a sensible time to make use of a `ParamSpec` to make it clearer a. what the intention is to the user and b. get some early warning from the type checker if you're using it wrong.

```suggestion
    plan: Callable[P, MsgGenerator] | None = None,
    plan_params: tuple[Unpack[P.args]] | None = None,  # this part I'm not 100% on the syntax!
```